pkg/bgpv1/agent/controller.go

```diff
@@ -24,16 +26,23 @@ import (
 	"github.com/cilium/cilium/pkg/logging"
 	"github.com/cilium/cilium/pkg/logging/logfields"
 	"github.com/cilium/cilium/pkg/option"
+	"github.com/cilium/cilium/pkg/promise"
 )
 
 var (
 	log = logging.DefaultLogger.WithField(logfields.LogSubsys, "bgp-control-plane")
+
+	// resolver and promise for the BGP controller, resolver is set during controller construction.
+	controllerResolver, controllerPromise = promise.New[*Controller]()
```

Comment:
Can't we avoid introducing a global object here? This object is out of the DI tree, so cannot get the benefit of Hive/Cell infrastructure.

Reply:
I am not sure if there is a way to do this without re-adding bgp controller dependency back into daemon. 

I was following the existing pattern from [metrics subsystem](https://github.com/cilium/cilium/blob/main/pkg/metrics/metrics.go#L267). Here registry is also declared globally and http calls directly into metrics package like [metrics.DumpMetrics](https://github.com/cilium/cilium/blob/main/pkg/metrics/metrics.go#L1517). 

As a middle ground, here is alternative proposal. To create global promise but hook it with hive by returning promise from controller init. Code : https://gist.github.com/harsimran-pabla/9935f0f572943b028187f9ef06324008

Let me know if you'd prefer to leave it as is, or use alternative approach.

Reply:
Updated PR to remove BGP API handlers from daemon and moving them to bgp control plane. 
(thanks @joamaki for providing solution on this.)